web/src/components/ak-wizard-main/AkWizard.ts

```diff
@@ -80,36 +98,60 @@ export class AkWizard<D, Step extends WizardStep = WizardStep>
      * You should still consider overriding this if you need to consider details like "Is the step
      * requested valid?"
      */
-    handleNav(stepId: number | undefined) {
-        if (stepId === undefined || this.steps[stepId] === undefined) {
-            throw new Error(`Attempt to navigate to undefined step: ${stepId}`);
+    navigateTo(stepId: string) {
+        const nextstep = this.steps.find((step) => step.id === stepId);
+        if (!nextstep) {
+            throw new Error(`Navigation request for a step that does not exist: ${stepId}`);
         }
-        this.currentStep = stepId;
+        this.currentStep = nextstep.id;
         this.requestUpdate();
     }
 
     close() {
         throw new Error("This function must be overridden in the child class.");
     }
 
+    onClose(_event: WizardCloseEvent) {
+        this.close();
+    }
+
     /**
      * This is where all the business logic and special cases go. The Wizard Controller intercepts
      * updates tagged `ak-wizard-update` and forwards the event content here. Business logic about
      * "is the current step valid?" and "should the Next button be made enabled" are controlled
      * here. (Any step implementing WizardStep can do it anyhow it pleases, putting "is the current
      * form valid" and so forth into the step object itself.)
      */
-    handleUpdate(_detail: D) {
+    onUpdate(_event: WizardUpdateEvent<D>) {
         throw new Error("This function must be overridden in the child class.");
     }
 
+    onNavigation(event: WizardNavigationEvent) {
```

Comment:
ts-pattern allows for some amazing narrowing-by-types, and using it here is a huge win.